extensions/notebook/src/jupyter/jupyterServerManager.ts

```diff
@@ -154,9 +178,25 @@ export class LocalJupyterServerManager implements nb.ServerManager, vscode.Dispo
 }
 
 export class ServerInstanceFactory {
+	private _instances: PerDriveServerInstance[] = [];
+
+	public checkIfInstanceExists(options: IInstanceOptions): IServerInstance | undefined {
+		let index = this._instances.findIndex(e => e.getNotebookDirectory() === options.notebookDirectory);
+		if (index > -1) {
+			if (this._instances[index] && this._instances[index].isStarted) {
+				this._instances[index].incrementAttachedEditorCount();
+				return this._instances[index];
+			} else {
+				this._instances.splice(index);
```

Comment:
What are you trying to do here? This is going to remove all instances after the found index too. I'm not even sure why you're removing instances in the first place?

Reply:
Heh, this is no longer needed.